### PR TITLE
Adds dedupe

### PR DESCRIPTION
**Pull request by jaihindhreddy**

### CODE REVIEW

File: clj/seqs.py [hunk 1]
```diff
@@ -655,6 +655,23 @@ def tree_seq(has_branch, get_children, root):
...
 
 
+def dedupe(coll):
+    """
+    Returns a generator of the elements of coll with consecutive duplicates removed.
+    """
+    coll = iter(coll)
+    try:
+        prev = next(coll)
+    except StopIteration:
```
Comment (bfontaine) [inline clj/seqs.py]: You can replace the try/except + a loop with a single loop:
```python3
def dedupe(coll):
    """
    ...
    """
    initial = True
    prev = None

    for e in coll:
        if initial or e != prev:
            initial = False
            yield e

        prev = e
```

**Comment by bfontaine** [comment]
Thank you @jaihindhreddy, that’s a perfect PR, complete with documentation and tests. I’ve just added one comment on the code, which can be simplified.

File: clj/seqs.py (hunk 1 above)
Comment (jaihindhreddy) [reply clj/seqs.py]: I didn't want to check that flag each time around the loop. Do you think that isn't a big enough deal?

Comment (bfontaine) [reply clj/seqs.py]: No, I don’t think that’s a big deal.

Comment (jaihindhreddy) [reply clj/seqs.py]: Done.

**Comment by bfontaine** [comment]
Merged; thanks again! :tada:

**Comment by bfontaine** [comment]
clj 0.1.5 is out, with `dedupe` in it: https://pypi.org/project/clj/0.1.5/

**Pull request merged**